File: 2178-maximum-split-of-positive-even-integers/2178-maximum-split-of-positive-even-integers.java
class Solution {
    public List<Long> maximumEvenSplit(long finalSum) {
        List<Long> list = new ArrayList<>();
        if(finalSum % 2 != 0) return list;
        long curr = 0;
        for(long i = 2; i <= finalSum - curr; i+=2) {
            curr += i;
            list.add(i);
            
        }
        
        if(finalSum > curr) {
            list.set(list.size() - 1, list.get(list.size() -1) + (finalSum - curr));
        }
        return list;
    }
}

File: 2178-maximum-split-of-positive-even-integers/2178-maximum-split-of-positive-even-integers.py
class Solution:
    def maximumEvenSplit(self, finalSum: int) -> List[int]:
        res = []
        if finalSum % 2 != 0:
            return res
        even = 2
        while finalSum >= even:
            finalSum -= even
            res.append(even)
            even += 2
        if finalSum > 0:
            last = res.pop() + finalSum
            res.append(last)
           
        return res
            